agent v6 python version management (#13347)
DataDog/documentation: content/en/agent/guide/agent-v6-python-3.md

However, there may be a case where you wish to continue using Agent v6 while updating to Python 3. Starting with Datadog Agent v6.14.0, Agent v6 integrates both Python 2 and Python 3 runtimes. This means that Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.

## Use Python 3 with Datadog Agent v6

By default, the Agent v6 uses the Python 2 runtime. Below are instructions for how to configure Agent v6 to use the Python 3 runtime:

- [Host Agent](#host-agent)
- [Container Agent](#container-agent)
  - [Helm](?tab=helm#container-agent)
  - [Datadog Operator](?tab=datadogoperator#container-agent)
  - [DaemonSet](?tab=daemonset#container-agent)
- [Deployment tools](#deployment-tools)
  - [Chef](?tab=chef#deployment-tools)
  - [Puppet](?tab=puppet#deployment-tools)
  - [Ansible](?tab=ansible#deployment-tools)

This configuration is not supported for the Azure VM Extension.

### Host Agent

1. Set the `python_version` configuration option in your [`datadog.yaml`][2] configuration file:

    ```yaml
    python_version: 3
    ```

2. [Restart the Agent][3].

> Alternatively, you can set the environment variable `DD_PYTHON_VERSION` to `2` or `3` to choose which Python runtime is used. Environment variables take precedence over configuration options in `datadog.yaml`. That is, if you set the environment variable `DD_PYTHON_VERSION`, the `python_version` option in `datadog.yaml` is ignored.

Reviewer: ```suggestion
Alternatively, specify which Python runtime you want to use by setting the `DD_PYTHON_VERSION` environment variable to `2` or `3`. Environment variables take precedence over configuration options in `datadog.yaml`. For example, by setting the `DD_PYTHON_VERSION` environment variable, the `python_version` option in `datadog.yaml` is ignored.
```

This is an Agent-wide configuration option. **All Python checks launched by an Agent use the same Python runtime**.

### Container Agent

Datadog provides Agent container images for Python 2 and Python 3. 

> * Image tags starting with `6.`, like `6.34.0` or `6.34.0-jmx` are images containing the Python 2 runtime.
> * Image tags starting with `7.`, like `7.34.0` or `7.34.0-jmx` are images containing the Python 3 runtime.

Reviewer: ```suggestion
* Image tags starting with `6.`, like `6.34.0` or `6.34.0-jmx`, are images containing the Python 2 runtime.
* Image tags starting with `7.`, like `7.34.0` or `7.34.0-jmx`, are images containing the Python 3 runtime.
```

clusterChecksRunner:
  image:
    repository: public.ecr.aws/datadog/agent
```

To set the Agent to a specific version, set `agents.image.tag` and  `clusterChecksRunner.image.tag`. All image tags starting with `7.*` embed the Python 3 runtime.

```yaml
agent:
  image:
    tag: 7.34.0

clusterChecksRunner:
  image:
    tag: 7.34.0
````

You can use both options at the same time.

```yaml
agent:
  image:
    repository: public.ecr.aws/datadog/agent
    tag: 7.34.0

clusterChecksRunner:
  image:
    repository: public.ecr.aws/datadog/agent
    tag: 7.34.0
```

[1]:https://artifacthub.io/packages/helm/datadog/datadog/

{{% /tab %}}
{{% tab "Datadog Operator" %}}
By default, the [Datadog Operator][1] uses an `agent:7.*.*` image that embeds the Python 3 runtime.

If the image information is not specified in the `DatadogAgent` resource, the Operator deploys a Python 3 Datadog Agent image.

If you have previously pinned the image version:

```yaml
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  # ...
  agent:
    # ...
    imageConfig:
      tag: 6.33.0
  clusterChecksRunner:
    # ...
    imageConfig:
      tag: 6.33.0
```

> or using `imageConfig.name`:

Reviewer: ```suggestion
or you are using `imageConfig.name`:
```
